README.md

```diff
@@ -1,3 +1,288 @@
+```
+
+### Automatic Registration
+
+```csharp
+builder.Services.AddSimpleTransit(options =>
+{
+    // Register all handlers and consumers from specified assembly
+    options.RegisterServicesFromAssemblyContaining<Program>();
+    
+    // Or from a specific assembly
+    options.RegisterServicesFromAssembly(typeof(MyHandler).Assembly);
+    
+    // With optional filtering
+    options.RegisterServicesFromAssembly(
+        assembly, 
+        type => type.Namespace?.StartsWith("MyApp.Handlers") == true);
+});
+```
+
+## Advanced Scenarios
+
+### Multiple Handlers
+Multiple notification handlers can be registered for the same message type:
+
+```csharp
+public class EmailNotificationHandler : INotificationHandler<PersonCreated>
+{
+    public async Task HandleAsync(PersonCreated message, CancellationToken cancellationToken)
+    {
+        // Send email
+    }
+}
+
+public class AuditNotificationHandler : INotificationHandler<PersonCreated>
+{
+    public async Task HandleAsync(PersonCreated message, CancellationToken cancellationToken)
+    {
+        // Log to audit system
+    }
+}
+```
+
+### Error Handling
+SimpleTransit propagates exceptions from handlers to allow for proper error handling:
+
+```csharp
+try
+{
+    await _notificationPublisher.NotifyAsync(message);
+}
+catch (Exception ex)
+{
+    // Handle errors from any of the notification handlers
+    _logger.LogError(ex, "Error processing notification");
+}
+```
+
+### Scoped Services
+SimpleTransit properly handles service scoping, especially in ASP.NET Core applications:
+
+```csharp
+public class DatabaseHandler(MyDbContext context) : INotificationHandler<PersonCreated>
+{
+    public async Task HandleAsync(PersonCreated message, CancellationToken cancellationToken)
+    {
+        // Use scoped DbContext safely
+        context.People.Add(new Person { Name = message.FirstName });
+        await context.SaveChangesAsync(cancellationToken);
+    }
+}
+```
+
+## Best Practices
+
+1. **Choose the Right Pattern**: Use notifications for immediate actions, messages for background processing
+2. **Keep Handlers Focused**: Each handler should have a single responsibility
+3. **Handle Errors Gracefully**: Always consider error scenarios in your handlers
+4. **Use Cancellation Tokens**: Support cancellation for long-running operations
+5. **Leverage Dependency Injection**: Take advantage of scoped services for database operations
+6. **Consider Performance**: Notifications are executed synchronously, so keep them fast
+
+## Samples
+
+A complete sample application is available in the [`samples/SimpleTransitSample`](samples/SimpleTransitSample) directory, demonstrating:
+
+- Web API integration
+- Both notification and message patterns
+- Service registration and configuration
+- Real-world usage scenarios
+
+To run the sample:
+
+```bash
+cd samples/SimpleTransitSample
+dotnet run
+```
+
+Then navigate to the Swagger UI at `https://localhost:5001/swagger` to test the API endpoints.
```

Comment:
5001 isn't the correct port. Change the text and specifies to navigate to the URL specified in the console.